### DISCUSSION TITLE
@apply not working (for me) as intended?

### DISCUSSION THREAD

**Discussion by whitigol** [tailwindlabs/tailwindcss#10131] [Help]
Currently working on a project as a first time user of tailwind. I'm trying to create a badge classes with SCSS, using @apply, and it's not working as intended.

Here's my `main.scss`:
```scss
@tailwind base;
@tailwind components;
@tailwind utilities;

.badge {
    @apply text-xs inline-block py-1 px-2.5 leading-none text-center whitespace-nowrap align-baseline font-bold;
    
    &.badge-primary {
        @apply bg-blue-500 text-white;
    }
    &.badge-secondary {
        @apply bg-purple-500 text-white;
    }
    &.badge-success {
        @apply bg-green-500 text-white;
    }
    &.badge-danger {
        @apply bg-red-500 text-white;
    }
    &.badge-warning {
        @apply bg-yellow-500 text-white;
    }
    &.badge-info {
        @apply bg-blue-500 text-white;
    }
    &.badge-light {
        @apply bg-gray-500 text-white;
    }
    &.badge-dark {
        @apply bg-gray-800 text-white;
    }
}
```

I can confirm that tailwind **is** working inside my project, as I'm using other classes on the HTML that are working as intended. So I'm not sure what the issue could be.

Thanks in advance for any assistance that can be provided!

**Comment by wongjn** [comment]
Could you give more details about how it does not work as intended? How do you intend it to work? The code looks fine – do you have a reproduction you could give for us to look at?

**Reply by wongjn** [reply]
> Sorry for such a late reply. So for some more details, in the browser, I get an error on the custom classes that reads "Unknown property name" when using @apply. I can confirm that tailwind is working in my project, because when using other classes, it compiles, and displays properly on the page. And when I am using @apply in the scss, it compiles and creates the used styles. But they don't appear on the browser at all.[^1]

> in the browser, I get an error on the custom classes that reads "Unknown property name" when using `@apply`

> And when I am using `@apply` in the scss, it compiles and creates the used styles. But they don't appear on the browser at all.

These statements seem contradictory – if <q>they don't appear on the browser at all<q> (where <q>they</q>) meaning the custom CSS rules, then:
- you would not be getting the "Unknown property name" error in the browser.
- Scss does *not* <q>compiles and creates the used styles</q> since if Scss (and Tailwind) did, they *would* <q>appear on the browser at all</q>.

Regardless, it seems like there is some misconfiguration or a bug in the build pipeline and we cannot assist any further without a reproduction for us to look at.

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/10131#discussioncomment-4494591

**Comment by whitigol** [comment]
Sorry for such a late reply. So for some more details, in the browser, I get an error on the custom classes that reads "Unknown property name" when using @apply. I can confirm that tailwind is working in my project, because when using other classes, it compiles, and displays properly on the page. And when I am using @apply in the scss, it compiles and creates the used styles. But they don't appear on the browser at all.

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/10131#discussioncomment-4496788 to keep the reply chain together.
